src/components/UI/AttachmentLink/AttachmentLink.tsx

```diff
@@ -57,21 +57,47 @@ export const getExtension = (mimeType: string | undefined): React.ReactNode => {
   }
 };
 
+//sending message to child window
+const openTab = (data: string) => {
+  const childWin = window.open(`${window.location}/#verify-documents`, "_blank");
+  window.addEventListener(
+    "message",
+    (event) => {
+      if (event.data.type == "READY" && childWin) {
+        childWin.postMessage(
+          {
+            type: "LOAD_DOCUMENT",
+            payload: data,
+          },
+          `${window.location.href}`
+        );
+      }
+    },
+    false
+  );
+};
+
+const isOpenAttestationFile = (decodedData: string) => {
+  try {
+    const decodedJson = JSON.parse(decodedData);
+    const unwrappedDocument = getData<WrappedDocument<OriginalDocumentProps>>(decodedJson);
+    if (!unwrappedDocument) throw new Error("File is not OA document"); //non-OA document returns undefined
+    return true;
+  } catch (e) {
+    error("decode data not json: " + e);
+    return false;
+  }
+};
+
 export const AttachmentLink = ({ filename, data, type, path }: AttachmentLinkProps) => {
   let filesize = "0";
-  let redirectLink = "";
+  let canOpenFile = false;
   const hasBase64 = !!(data && type);
   const downloadHref = hasBase64 ? `data:${type};base64,${data}` : path || "#";
   if (data) {
```

Comment:
Actually... will there ever be a case where there is no `data` here?

Reply:
As the AttachmentLinkProps is 
```js
export interface AttachmentLinkProps {
  filename: string;
  data?: string;
  type?: string;
  path?: string;
}
```
there is a chance that data is a string or undefined. But I can enforce the type to be `data:string` since data should always be present for the AttachmentLink (there is a ResourceLink which is used for the Media and Resources page which has the `details.url` for downloads from a specified path).

This is a current outdated testcase in AttachmentLink (as links to resources is now handled by ResourceLink):
```js
  it("should render download link with direct asset correctly", () => {
    const container = render(
      <AttachmentLink
        filename="TradeTrust Tech Webinar 2.pdf"
        path="/static/images/webinar/tradetrust-tech-webinar-2.pdf"
      />
    );
    expect(container.getByText("Download")).not.toBeNull();
  });
```

Reply:
I can:

1. Edit the prop types for `data` to enforce presence, edit the tests and remove the `if(data)`.
or
2. Leave the `if(data)` check

Reply:
yea if its always present then really dont need that liao